src/Compilers/Core/Portable/CommandLine/CommandLineArguments.cs

```diff
@@ -524,12 +524,14 @@ public IEnumerable<AnalyzerReference> ResolveAnalyzerReferences(IAnalyzerAssembl
                 }
             }
 
-            // All analyzer references are registered now, we can start loading them:
+            // If we are skipping analyzers, ensure that we only add suppressors.
+            bool shouldIncludeAnalyzer(DiagnosticAnalyzer analyzer) => !skipAnalyzers || analyzer is DiagnosticSuppressor;
```

Comment:
Nit: Consider putting this at the end of the function, or just inlining as a lambda.

Reply:
Yes, please move local function toward the end of the method.

Reply:
Done - sorry for the delay. @jcouv can you please take another look?